File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    divisor = []
    for i in range(1,int((yellow)**(0.5))+1):
        if yellow % i == 0:
            divisor.append(i)
    
    for div in divisor:
        y = div+2
        x = (yellow//div)+2
        if (x*y) == (brown+yellow):
            return [x,y]

File: 프로그래머스/2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = {};
        int x = 2;
        int y = 0;
        int total = brown+yellow;
        while (true) {
            x += 1;
            if (total%x == 0){
                y = total/x;
                if ((x>=y) && (x-2)*(y-2) == yellow) break;
                
            } else continue;
        }
        return new int[]{x,y};
    }
}